src/components/Table/GridTable.tsx

```diff
@@ -333,32 +344,34 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
           columns.map((c) => applyRowFn(c, row)).some((maybeContent) => matchesFilter(maybeContent, filter)),
         );
       // Even if we don't pass the filter, one of our children might, so we continue on after this check
+      let isCard = false;
       if (matches) {
-        nestedCards && nestedCards.beginRow(row);
+        isCard = nestedCards && nestedCards.maybeOpenCard(row);
```

Comment:
If this row doesn't have a card style, then we return false, which tells us later that we don't need to close the card.